src/features/timebank/timebank-API-provider.ts

```diff
@@ -54,23 +57,26 @@ namespace TimeBankApiProvider {
    */
   export const getTotalTimeEntries = async (timePeriod: TimePeriod, person: PersonDto, year: number, week: number): Promise<WeeklyCombinedData> => {
```

Comment:
I cannot make a suggestion so here is the suggested function:
```
    try {
      const { body } = await client.timebankControllerGetTotal(person.id.toString(), timePeriod);
      if (!body.length) throw new Error("Error while loading total time entries");

      const filteredWeeks = body.filter(timePeriod => timePeriod.id.year === year && timePeriod.id.week === week);
      if (filteredWeeks.length !== 1) throw new Error("Found more than one time period for given year and week");

      const selectedWeek = filteredWeeks[0];
      const { firstName, lastName } = person;
      const combinedName = `${firstName} ${lastName}`;

      return {
        selectedWeek: selectedWeek,
        name: combinedName,
        firstName: person.firstName,
        personId: person.id,
        expected: person.monday
      };
    } catch (error) {
      console.error(error);
      Promise.reject(error);
    }
```

Reply:
Few things have changed:
- You can one-line if checks
- It is better to use shallow code structure which means that instead of wrapping blocks of code inside if checks you should make if checks return out from the function
- Take filtered list into own variable and check for length --> You can assure that there is only one entry before using the array index selector.
- Remove return from Promise.reject